src/server/server_family.cc

```diff
@@ -1988,17 +1988,20 @@ void ServerFamily::Role(CmdArgList args, ConnectionContext* cntx) {
 
   } else {
     auto replica_ptr = replica_;
+    CHECK(replica_ptr);
     Replica::Info rinfo = replica_ptr->GetInfo();
     (*cntx)->StartArray(4);
     (*cntx)->SendBulkString("replica");
     (*cntx)->SendBulkString(rinfo.host);
     (*cntx)->SendBulkString(absl::StrCat(rinfo.port));
-    if (rinfo.sync_in_progress) {
+    if (rinfo.full_sync_done) {
+      (*cntx)->SendBulkString("stable_sync");
+    } else if (rinfo.full_sync_in_progress) {
       (*cntx)->SendBulkString("full_sync");
-    } else if (!rinfo.master_link_established) {
-      (*cntx)->SendBulkString("connecting");
+    } else if (rinfo.master_link_established) {
```

Comment:
Consider 
disconnected -> connecting -> full_sync -> stable_sync
or
connecting -> handshaking -> full_sync -> stable_sync

Reply:
I chose `preparation` because it's the term used in `ReplicaRoleInfo::ReplicaRoleInfo` and in `SyncState::PREPARATION`. Do you still prefer handshaking?

Reply:
Nope :)